spec/performable_mailer_integration_spec.rb

```diff
@@ -22,15 +26,23 @@
         TestMailer.delay.reticulate(@arg)
       end.to change(Delayed::Job, :count).by(1)
       @arg.destroy
-      expect_any_instance_of(Mail::Message).to receive(:deliver).never
+      if ActionMailer::VERSION::STRING >= '4.2.0'
+        expect_any_instance_of(ActionMailer::MessageDelivery).to receive(:deliver_now).never
+      else
+        expect_any_instance_of(Mail::Message).to receive(:deliver).never
+      end
       expect(Delayed::Worker.new.work_off).to eq([1, 0])
     end
     it "ignores deleted models when find doesn't raise an error" do
       expect(TestModel).to receive(:find).with(@arg.id.to_s).and_return(nil)
       expect do
         TestMailer.delay.reticulate(@arg)
       end.to change(Delayed::Job, :count).by(1)
-      expect_any_instance_of(Mail::Message).to receive(:deliver).never
+      if ActionMailer::VERSION::STRING >= '4.2.0'
+        expect_any_instance_of(ActionMailer::MessageDelivery).to receive(:deliver_now).never
+      else
+        expect_any_instance_of(Mail::Message).to receive(:deliver).never
+      end
```

Comment:
This is just different syntax and a little redundant, but one thing we could do is the following:

```ruby
it 'ignores deleted models when find raises Mongoid::Errors::DocumentNotFound', if: ActionMailer::VERSION::STRING >= '4.2.0' do
  expect { TestMailer.delay.reticulate(@arg) }.to change(Delayed::Job, :count).by(1)
  @arg.destroy

  expect_any_instance_of(ActionMailer::MessageDelivery).to receive(:deliver_now).never
  expect(Delayed::Worker.new.work_off).to eq([1, 0])
end

it 'ignores deleted models when find raises Mongoid::Errors::DocumentNotFound', if: ActionMailer::VERSION::STRING < '4.2.0' do
  expect { TestMailer.delay.reticulate(@arg) }.to change(Delayed::Job, :count).by(1)
  @arg.destroy

  expect_any_instance_of(Mail::Message).to receive(:deliver).never
  expect(Delayed::Worker.new.work_off).to eq([1, 0])
end
```